api/src/test/java/io/kafbat/ui/OpenLdapPIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.kafbat.ui.container.OpenLdapContainer;
+import io.kafbat.ui.model.AuthenticationInfoDTO;
+import io.kafbat.ui.model.ResourceTypeDTO;
+import io.kafbat.ui.model.UserPermissionDTO;
+import java.util.List;
+import java.util.Objects;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.web.reactive.function.BodyInserters;
+
+@SpringBootTest
+@ActiveProfiles("rbac-ldap")
+@AutoConfigureWebTestClient(timeout = "60000")
+@ContextConfiguration(initializers = {OpenLdapPIntegrationTest.Initializer.class})
+class OpenLdapPIntegrationTest {
+  private static final String SESSION = "SESSION";
+  private static final OpenLdapContainer LDAP_CONTAINER = new OpenLdapContainer();
+
+  @Autowired
+  private WebTestClient webTestClient;
+
+  @DynamicPropertySource
+  static void neo4jProperties(DynamicPropertyRegistry registry) {
+    registry.add("spring.ldap.urls", LDAP_CONTAINER::getLdapUrl);
+  }
+
+  @BeforeAll
+  static void setup() {
+    LDAP_CONTAINER.start();
+  }
+
+  @AfterAll
+  static void shutdown() {
+    LDAP_CONTAINER.stop();
+  }
+
+  @Test
+  public void testUserPermissions() {
+    AuthenticationInfoDTO info = authenticationInfo("johndoe");
+
+    assertNotNull(info);
+    assertTrue(info.getRbacEnabled());
+    System.out.println("info = " + info);
+    List<UserPermissionDTO> permissions = info.getUserInfo().getPermissions();
+    assertFalse(permissions.isEmpty());
+    assertTrue(permissions.stream().anyMatch(permission ->
+        permission.getClusters().contains(LOCAL) && permission.getResource() == ResourceTypeDTO.TOPIC));
+    assertEquals(permissions, authenticationInfo("johnwick").getUserInfo().getPermissions());
+    assertEquals(permissions, authenticationInfo("jacksmith").getUserInfo().getPermissions());
+  }
+
+  @Test
+  public void testDirectUserPermissions() {
+    AuthenticationInfoDTO info = authenticationInfo("jacksmith");
+
+    assertNotNull(info);
+    assertTrue(info.getRbacEnabled());
+    System.out.println("info = " + info);
+    List<UserPermissionDTO> permissions = info.getUserInfo().getPermissions();
+    assertFalse(permissions.isEmpty());
```

Comment:
this particular test looks redundant -- in `testUserPermissions` we've asserted that permissions of `jacksmith` are the same as `johndoe`'s **and** have a topic permissions